File: src/asyn.cc

#include <node.h>
#include "asyn_class.h"



using namespace v8;

void Init(Handle<Object> target) {

  HandleScope scope;

  target->Set(String::NewSymbol("asyn"),
           FunctionTemplate::New(Asyn::do_asyn)->GetFunction());

  target->Set(String::NewSymbol("asyn_thread"),
           FunctionTemplate::New(Asyn::do_asyn_thread)->GetFunction());

  target->Set(String::NewSymbol("asyn_pool"),
           FunctionTemplate::New(Asyn::do_asyn_pool)->GetFunction());

  target->Set(String::NewSymbol("sync"),
           FunctionTemplate::New(Asyn::do_sync)->GetFunction());

  target->Set(String::NewSymbol("sync_thread"),
           FunctionTemplate::New(Asyn::do_sync_thread)->GetFunction());


}

NODE_MODULE(asyn, Init)

File: lib/asyn.js
var asyn_obj = require('../build/Release/asyn.node');

var check = function(t, n, func){
	if(!isFinite(t)) throw("arg[0] must a number")
	if(!isFinite(n)) throw("arg[1] must a number")
	if("function" !== typeof func) throw("arg[2] must a function")
	return true;
}

var asyn = function(t,i,f){
	check(t,i,f) && asyn_obj.asyn(t,i,f);
}
var asyn_thread = function(t,i,f){
	check(t,i,f) && asyn_obj.asyn_thread(t,i,f);
}
var asyn_pool = function(t,i,f){
	check(t,i,f) && asyn_obj.asyn_pool(t,i,f);
}
var sync = function(t,i,f){
	check(t,i,f) && asyn_obj.sync(t,i,f);
}
var sync_thread = function(t,i,f){
	check(t,i,f) && asyn_obj.sync_thread(t,i,f);
}


module.exports = {
	asyn:asyn,
	asyn_thread:asyn_thread,
    asyn_pool:asyn_pool,
	sync:sync,
	sync_thread:sync_thread
};
